File: app/src/main/java/com/chopin/chopin/fragments/MyOfferList.java
package com.chopin.chopin.fragments;

import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.chopin.chopin.API.API;
import com.chopin.chopin.R;
import com.chopin.chopin.adapters.RVAdapter;
import com.chopin.chopin.models.Offer;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MyOfferList extends Fragment{

    private API.APIInterface _api;
    private ArrayList<Offer> offers;
    private RVAdapter adapter;
    @BindView(R.id.me_offer_list) RecyclerView mRecyclerView;
    @BindView(R.id.swipe) SwipeRefreshLayout swipe;


    public MyOfferList() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        _api = API.getClient();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_me_offer_list, container, false);
        ButterKnife.bind(this,v);
        return v;
    }

    @Override
    public void onViewCreated(final View view, Bundle savedInstanceState) {
        get();
        swipe.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                Log.d("onRefresh", "calling get()");
                get();
                adapter.notifyDataSetChanged();
                swipe.setRefreshing(false);
            }
        });
    }

    public void get(){
        Call<List<Offer>> query = _api.getMyOffers();
        offers = new ArrayList<>();
        query.enqueue(new Callback<List<Offer>>() {

            @Override
            public void onResponse(Call<List<Offer>> call, Response<List<Offer>> response) {
                if (response.isSuccessful()) {
                    Log.d("onResponse", "got " + response.body().size());
                    offers.addAll(response.body());
                    adapter = new RVAdapter(offers,getActivity());
                    mRecyclerView.setAdapter(adapter);
                }
            }

            @Override
            public void onFailure(Call<List<Offer>> call, Throwable t) {;
                Snackbar.make(getView(), "Connection problem", Snackbar.LENGTH_INDEFINITE).show();
            }

        });
        mRecyclerView.setHasFixedSize(true);
        // use a linear layout manager
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(this.getContext());
        mRecyclerView.setLayoutManager(mLayoutManager);
    }

}

File: app/src/main/java/com/chopin/chopin/fragments/MyOfferList.kt
package com.chopin.chopin.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.widget.SwipeRefreshLayout
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import bindView
import butterknife.ButterKnife
import com.chopin.chopin.ConnectionHandler
import com.chopin.chopin.R
import com.chopin.chopin.adapters.RVAdapter
import com.chopin.chopin.models.Offer
import java.util.*

class MyOfferList : Fragment() {

    private var offers: ArrayList<Offer>? = null
    private var adapter: RVAdapter? = null
    internal val mRecyclerView: RecyclerView by bindView(R.id.me_offer_list)
    val swipe : SwipeRefreshLayout by bindView(R.id.swipe)
    private var connectionHandler: ConnectionHandler = ConnectionHandler()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        offers = ArrayList<Offer>()
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val v = inflater!!.inflate(R.layout.fragment_me_offer_list, container, false)
        ButterKnife.bind(this, v)
        return v
    }

    override fun onStart() {
        super.onStart()
        mRecyclerView?.setHasFixedSize(true)
        // use a linear layout manager
        val mLayoutManager = LinearLayoutManager(this.context)
        mRecyclerView?.layoutManager = mLayoutManager
        offers = connectionHandler.myOfferFromServer
        adapter = RVAdapter(offers, activity)
        mRecyclerView!!.adapter = adapter
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {

        swipe?.setOnRefreshListener {
            Log.d("onRefresh", "calling get()")
            offers = connectionHandler.myOfferFromServer
            adapter!!.notifyDataSetChanged()
            swipe!!.isRefreshing = false
        }
    }

}// Required empty public constructor
